soundconverter/interface/batch.py

```diff
@@ -64,7 +64,7 @@ def use_memory_gsettings(options):
     gio_settings = Gio.Settings.new_with_backend('org.soundconverter', backend)
     set_gio_settings(gio_settings)
 
-    gio_settings.set_boolean('delete-original', False)
+    gio_settings.set_boolean('delete-original', options['delete-original'])
```

Comment:
Thanks for the change, but now tests are not passing. Try

```suggestion
    gio_settings.set_boolean('delete-original', options.get('delete-original', False))
```

Also, feel free to add a new test. You could look at `test_set_delete_original_false` in `integration.py` and make a copy of it that tests the `-D` argument. And then use something like `self.assertFalse(os.path.isfile(...))` to check if it correctly deleted the source file.

Reply:
Done.